content/developers/dogstatsd/data_types.md

```diff
@@ -0,0 +1,309 @@
+
+Learn more about the [Histogram type in the Metrics documentation][6].
+
+### Timers
+
+Timers in DogStatsD are an implementation of Histograms (not to be confused with timers in the standard StatsD). They measure timing data _only_, such as the amount of time a section of code takes to execute, such as the time it takes to render a web page. In Python, you can create timers with a decorator:
+
+```python
+
+from datadog import statsd
+
+@statsd.timed('mywebsite.page_render.time')
+def render_page():
+  # Render the page...
+```
+
+or with a context manager:
+
+```python
+
+from datadog import statsd
+
+def render_page():
+  # First some stuff we don't want to time
+  boilerplate_setup()
+
+  # Now start the timer
+  with statsd.timed('mywebsite.page_render.time'):
+    # Render the page...
+```
+
+In either case, as DogStatsD receives the timer data, it calculates the statistical distribution of render times and sends the following metrics to Datadog:
+
+- `mywebsite.page_render.time.count` - the number of times the render time was sampled
+- `mywebsite.page_render.time.avg` - the average render time
+- `mywebsite.page_render.time.median` - the median render time
+- `mywebsite.page_render.time.max` - the maximum render time
+- `mywebsite.page_render.time.95percentile` - the 95th percentile render time
+
+Under the hood, DogStatsD treats timers as histograms. Whether you timers or histograms, you'll be sending the same data to Datadog.
+
+### Sets
+
+Sets are used to count the number of unique elements in a group, for example, the number of unique visitors to your site:
+
+For Python:
+```python
+
+def login(self, user_id):
+    # Log the user in ...
+    statsd.set('users.uniques', user_id)
+```
+
+For Ruby:
+```ruby
+def login(self, user_id)
+    # Log the user in ...
+    statsd.set('users.uniques', user_id)
+end
+```
+
+Learn more about the [Histogram type in the Metrics documentation][6].
+
+## Metric option: sample rates
+
+Since the overhead of sending UDP packets can be too great for some performance
+intensive code paths, DogStatsD clients support sampling,
+i.e. only sending metrics a percentage of the time. The following code sends
+a histogram metric only about half of the time:
+
+```python
+
+dog.histogram('my.histogram', 1, sample_rate=0.5)
+```
+
+Before sending the metric to Datadog, DogStatsD uses the `sample_rate` to
+correct the metric value, i.e. to estimate what it would have been without sampling.
+
+**Sample rates only work with counter, histogram, and timer metrics.**
+
+Learn more about the [Rates in the Metrics documentation][7].
+
+## Events
+
+DogStatsD can emit events to your Datadog event stream. For example, you may want to see errors and exceptions in Datadog:
```

Comment:
DogStatsD can emit events to your Datadog [event stream][ADD LINK TO EVENT STEAM], allowing you to see errors and exceptions in Datadog:

Reply:
https://app.datadoghq.com/event/stream